src/main/java/com/google/devtools/build/lib/skyframe/ConfiguredTargetFunction.java

```diff
@@ -314,6 +338,63 @@ public SkyValue compute(SkyKey key, Environment env)
                 getPrioritizedDetailedExitCode(causes)));
       }
 
+      Rule rule = null;
+      if (target instanceof Rule) {
+        rule = (Rule) target;
+      } else if (target instanceof OutputFile) {
+        rule = ((OutputFile) target).getAssociatedRule();
+      }
+
+      PlatformInfo platformInfo = unloadedToolchainContexts != null ? unloadedToolchainContexts.getTargetPlatform() : null;
+      Label platformLabel = platformInfo != null ? platformInfo.label() : null;
+
```

Comment:
Do you think this will work on aliases? https://github.com/bazelbuild/bazel/commit/1c3a2456c95fd19974a5b2bd33c5ebdb2b2277e4 fixed a related problem, but that only applies when the alias has a `select()`.

The key point is that aliases are some rare rule that don't opt into toolchain resolution: https://github.com/bazelbuild/bazel/blob/1c3a2456c95fd19974a5b2bd33c5ebdb2b2277e4/src/main/java/com/google/devtools/build/lib/rules/Alias.java#L86

That means, as I understand, that if the alias has no `select()`, then none of the toolchain logic kicks in, which means `platformInfo == null`, which would skip all this checking.

Can we verify that on a simple example alias? We historically had similar problems with filegroup but I think that was fixed (checks notes... yup: https://github.com/bazelbuild/bazel/commit/dfb70ea4cae2ffffb76e9741d86c96505a6d05ad)

Here's the full list: https://github.com/bazelbuild/bazel/search?q=%22useToolchainResolution%22. It looks like there are no other rule types to worry about. I'm bringing this point up since people use aliases and I want to protect us against "feature X works except for target Y" bugs as much as we can foresee them.

Reply:
That sounds fine to me. Honestly, I don't remember the full context for my previous comment (i.e. what I meant by saying "which would skip all this checking"? Do you still see anything relevant in that comment? If not I'm happy to resolve, especially since you're showing alias test coverage.

Reply:
Resolving sounds good to me. I am pretty sure if there _is_ an issue with this, it should be easy to resolve. That being said, I should make sure that this patch works as expected with  #15426.

Reply:
This patch together with #15426 still passes all the tests. Resolving this comment.